api/utils/keys/policy.go

```diff
@@ -34,49 +34,110 @@ const (
 	case PrivateKeyPolicyNone:
-		return nil
+		return true
 	case PrivateKeyPolicyHardwareKey:
-		if policy == PrivateKeyPolicyHardwareKey || policy == PrivateKeyPolicyHardwareKeyTouch {
-			return nil
-		}
+		return keyPolicy.IsHardwareKeyPolicy()
 	case PrivateKeyPolicyHardwareKeyTouch:
-		if policy == PrivateKeyPolicyHardwareKeyTouch {
-			return nil
-		}
+		return keyPolicy.isHardwareKeyTouchVerified()
+	case PrivateKeyPolicyHardwareKeyPIN:
+		return keyPolicy.isHardwareKeyPINVerified()
+	case PrivateKeyPolicyHardwareKeyTouchAndPIN:
+		return keyPolicy.isHardwareKeyTouchVerified() && keyPolicy.isHardwareKeyPINVerified()
 	}
-	return NewPrivateKeyPolicyError(p)
+
+	return false
+}
+
+// Deprecated in favor of IsRequiredPolicyMet.
+// TODO(Joerger): delete once reference in /e is replaced.
+func (requiredPolicy PrivateKeyPolicy) VerifyPolicy(keyPolicy PrivateKeyPolicy) error {
+	if !IsRequiredPolicyMet(requiredPolicy, keyPolicy) {
+		return NewPrivateKeyPolicyError(requiredPolicy)
+	}
+	return nil
 }
 
-// IsHardwareKeyVerified return true if this private key policy requires a hardware key.
-func (p PrivateKeyPolicy) IsHardwareKeyVerified() bool {
+// IsHardwareKeyPolicy return true if this private key policy requires a hardware key.
+func (p PrivateKeyPolicy) IsHardwareKeyPolicy() bool {
 	switch p {
-	case PrivateKeyPolicyHardwareKey, PrivateKeyPolicyHardwareKeyTouch:
+	case PrivateKeyPolicyHardwareKey,
+		PrivateKeyPolicyHardwareKeyTouch,
+		PrivateKeyPolicyHardwareKeyPIN,
+		PrivateKeyPolicyHardwareKeyTouchAndPIN:
 		return true
 	}
 	return false
 }
 
-// MFAVerified checks that meet this private key policy counts towards MFA verification.
+// MFAVerified checks that private keys with this key policy count as MFA verified.
+// Both Hardware key touch and pin are count as MFA verification.
 func (p PrivateKeyPolicy) MFAVerified() bool {
-	return p == PrivateKeyPolicyHardwareKeyTouch
+	return p.isHardwareKeyTouchVerified() || p.isHardwareKeyPINVerified()
+}
+
+func (p PrivateKeyPolicy) isHardwareKeyTouchVerified() bool {
+	switch p {
+	case PrivateKeyPolicyHardwareKeyTouch, PrivateKeyPolicyHardwareKeyTouchAndPIN:
+		return true
+	}
+	return false
+}
+
+func (p PrivateKeyPolicy) isHardwareKeyPINVerified() bool {
+	switch p {
+	case PrivateKeyPolicyHardwareKeyPIN, PrivateKeyPolicyHardwareKeyTouchAndPIN:
+		return true
+	}
+	return false
 }
 
 func (p PrivateKeyPolicy) validate() error {
 	switch p {
-	case PrivateKeyPolicyNone, PrivateKeyPolicyHardwareKey, PrivateKeyPolicyHardwareKeyTouch:
+	case PrivateKeyPolicyNone,
+		PrivateKeyPolicyHardwareKey,
+		PrivateKeyPolicyHardwareKeyTouch,
+		PrivateKeyPolicyHardwareKeyPIN,
+		PrivateKeyPolicyHardwareKeyTouchAndPIN:
 		return nil
 	}
 	return trace.BadParameter("%q is not a valid key policy", p)
 }
 
+// GetPolicyFromSet returns least restrictive policy necessary to meet the given set of policies.
+func GetPolicyFromSet(policies []PrivateKeyPolicy) (PrivateKeyPolicy, error) {
```

Comment:
Maybe: PolicyThatSatisfiesSet ?

Reply:
Wow, you have no clue how long I've been looking for the word "satisfied" to replace "met". Too bad I can't change the key policy error from "private key policy not met" to "private key policy not satisfied" due to the client dependency...